R/teal_slice.R

```diff
@@ -483,44 +520,51 @@ c.teal_slices <- function(...) {
   )
 }
 
-
-# format method for teal_slices
+# store method for `teal_slices` object
+#' @param x `teal_slices` object
+#' @param file a `character(1)` with the storage destination name
 #' @export
 #' @rdname teal_slice
 #' @keywords internal
 #'
-format.teal_slices <- function(x, show_all = FALSE, ...) {
-  res <- character(0)
-  for (i in seq_along(x)) {
-    ind <- names(x)[i]
-    if (is.null(ind)) ind <- sprintf("[[%d]]", i)
-    res <- append(res, ind)
-    res <- append(res, format(x[[i]], show_all = show_all, ...))
-  }
+store_filters <- function(x, file, ...) {
+  checkmate::assert_class(x, "teal_slices")
+  checkmate::assert_path_for_output(file, overwrite = TRUE, extension = "json")
 
-  includes <- attr(x, "include_varnames")
-  if (length(includes) > 0L) {
-    res <- append(res, "\nfilterable variables:")
-    for (i in seq_along(includes)) {
-      res <- append(res, sprintf(" $ %s: %s", names(includes)[i], toString(includes[[i]])))
-    }
-  }
+  cat(format(x, show_all = TRUE, center = FALSE), file = file)
+}
 
-  excludes <- attr(x, "exclude_varnames")
-  if (length(excludes) > 0L) {
-    res <- append(res, "\nnon-filterable variables:")
-    for (i in seq_along(excludes)) {
-      res <- append(res, sprintf(" $ %s: %s", names(excludes)[i], toString(excludes[[i]])))
-    }
-  }
+# format method for `teal_slices`
+#' @param x `teal_slice` object
+#' @param show_all `logical(1)` should parameters set to NULL be returned
+#' @param center `logical(1)` should the output be centered and trimmed
+#' @export
+#' @rdname teal_slice
+#' @keywords internal
+#'
+format.teal_slices <- function(x, show_all = FALSE, center = TRUE, ...) {
+  checkmate::assert_flag(show_all)
+  checkmate::assert_flag(center)
 
-  ct <- attr(x, "count_type")
-  res <- append(res, sprintf("\ncount type: %s", ct))
-  paste(c(res, ""), collapse = "\n")
+  x_format <- lapply(x, format, show_all = show_all, center = center)
+  paste0(c(unlist(x_format), extract_attrs(x)), collapse = '\n')
+}
+
+# attributes extraction for `teal_slices` object
+#' @param x `teal_slice` object
+#' @keywords internal
+#'
+extract_attrs <- function(x){
+  attributes <- attributes(x)
+  attributes$class <- NULL
+  paste0(
+    'attributes\n',
+    jsonlite::toJSON(attributes, pretty = TRUE, auto_unbox = TRUE)
+  )
```

Comment:
This can be moved to `format.teal_slices`, it does not need a separate function.

Reply:
This is arguable but I don't mind because format.teal_slices is shorter than I expected at the end of the day. When it was longer, it was easier to understand the logic of a function if it was an extra function. Thanks for pointing this in the final stage
a274cd631cf3ef2bda920002f34949eca407f107